packages/utils/src/mergeDeep.ts

```diff
@@ -17,6 +17,22 @@ export function mergeDeep<S extends any[]>(
   if (respectPrototype) {
     Object.setPrototypeOf(output, Object.create(Object.getPrototypeOf(target)));
   }
+  if (Array.isArray(target)) {
+    // Are they all arrays with the same length?
+    const allArrays = sources.every(
+      source => Array.isArray(source) && source.length === target.length,
+    );
+    if (allArrays) {
```

Comment:
We don't do anything if they don't all have the same length ?

Reply:
I put it behind the flag. So it depends. If they are arrays but not in the same length or respectArrayLength flag is disabled, the array is concatenated.